File: SuperLauncherCpp/src/ui/dialogs/LoginDialog.cpp
#include "LoginDialog.h"
#include <QMessageBox>

LoginDialog::LoginDialog(QWidget *parent)
    : QDialog(parent)
{
    setupUi();
}

void LoginDialog::setupUi()
{
    setWindowTitle("Вход в аккаунт");
    setFixedSize(380, 300);
    setStyleSheet("QDialog { background: #1E1E2E; }");

    auto *layout = new QVBoxLayout(this);
    layout->setContentsMargins(24, 24, 24, 24);
    layout->setSpacing(12);

    auto *title = new QLabel("Вход", this);
    title->setStyleSheet("color: white; font-size: 20px; font-weight: bold;");
    title->setAlignment(Qt::AlignCenter);
    layout->addWidget(title);

    m_username = new QLineEdit(this);
    m_username->setPlaceholderText("Логин или Email");
    m_username->setStyleSheet(
        "QLineEdit { background: rgba(255,255,255,0.08); color: white; "
        "border: 1px solid rgba(255,255,255,0.15); border-radius: 6px; "
        "padding: 10px; font-size: 13px; }");

    m_password = new QLineEdit(this);
    m_password->setPlaceholderText("Пароль");
    m_password->setEchoMode(QLineEdit::Password);
    m_password->setStyleSheet(
        "QLineEdit { background: rgba(255,255,255,0.08); color: white; "
        "border: 1px solid rgba(255,255,255,0.15); border-radius: 6px; "
        "padding: 10px; font-size: 13px; }");

    m_remember = new QCheckBox("Запомнить меня", this);
    m_remember->setStyleSheet("color: #AAAAAA;");

    m_loginBtn = new QPushButton("Войти", this);
    m_loginBtn->setStyleSheet(
        "QPushButton { background: #7C4DFF; color: white; border: none; "
        "border-radius: 6px; padding: 10px; font-size: 14px; }"
        "QPushButton:hover { background: #9C6DFF; }");
    connect(m_loginBtn, &QPushButton::clicked, this, [this]() {
        if (m_username->text().isEmpty() || m_password->text().isEmpty()) {
            QMessageBox::warning(this, "Ошибка", "Заполните все поля");
            return;
        }
        accept();
    });

    auto *cancelBtn = new QPushButton("Отмена", this);
    cancelBtn->setStyleSheet(
        "QPushButton { background: #555; color: white; border: none; "
        "border-radius: 6px; padding: 10px; font-size: 14px; }"
        "QPushButton:hover { background: #666; }");
    connect(cancelBtn, &QPushButton::clicked, this, &QDialog::reject);

    layout->addWidget(m_username);
    layout->addWidget(m_password);
    layout->addWidget(m_remember);
    layout->addWidget(m_loginBtn);
    layout->addWidget(cancelBtn);
}

QString LoginDialog::username() const { return m_username->text(); }
QString LoginDialog::password() const { return m_password->text(); }
bool LoginDialog::rememberMe() const { return m_remember->isChecked(); }

File: SuperLauncher/ui/dialogs/login_dialog.py
from PyQt6.QtWidgets import (QDialog, QVBoxLayout, QHBoxLayout, QLabel, QPushButton,
                              QLineEdit, QTabWidget, QWidget, QFormLayout, QCheckBox, QMessageBox)
from PyQt6.QtCore import Qt
from PyQt6.QtGui import QCursor

class LoginDialog(QDialog):
    def __init__(self, account_system, parent=None):
        super().__init__(parent)
        self.account = account_system
        self.user_data = None
        self.setWindowTitle("Вход / Регистрация")
        self.setFixedSize(400, 450)

        layout = QVBoxLayout(self)
        title = QLabel("SuperLauncher Аккаунт")
        title.setStyleSheet("font-size: 22px; font-weight: bold; color: #4facfe;")
        title.setAlignment(Qt.AlignmentFlag.AlignCenter)
        layout.addWidget(title)

        tabs = QTabWidget()

        login_tab = QWidget()
        fl = QFormLayout(login_tab)
        self.login_user = QLineEdit()
        self.login_user.setPlaceholderText("Имя или Email")
        fl.addRow("Логин:", self.login_user)
        self.login_pass = QLineEdit()
        self.login_pass.setEchoMode(QLineEdit.EchoMode.Password)
        self.login_pass.setPlaceholderText("Пароль")
        fl.addRow("Пароль:", self.login_pass)
        tabs.addTab(login_tab, "Вход")

        reg_tab = QWidget()
        rl = QFormLayout(reg_tab)
        self.reg_user = QLineEdit()
        self.reg_user.setPlaceholderText("Имя пользователя")
        rl.addRow("Имя:", self.reg_user)
        self.reg_email = QLineEdit()
        self.reg_email.setPlaceholderText("Email")
        rl.addRow("Email:", self.reg_email)
        self.reg_pass = QLineEdit()
        self.reg_pass.setEchoMode(QLineEdit.EchoMode.Password)
        self.reg_pass.setPlaceholderText("Пароль")
        rl.addRow("Пароль:", self.reg_pass)
        self.reg_pass2 = QLineEdit()
        self.reg_pass2.setEchoMode(QLineEdit.EchoMode.Password)
        self.reg_pass2.setPlaceholderText("Повторите пароль")
        rl.addRow("Подтверждение:", self.reg_pass2)
        tabs.addTab(reg_tab, "Регистрация")

        layout.addWidget(tabs)

        btns = QHBoxLayout()
        login_btn = QPushButton("Войти")
        login_btn.clicked.connect(self._login)
        btns.addWidget(login_btn)
        reg_btn = QPushButton("Зарегистрироваться")
        reg_btn.clicked.connect(self._register)
        btns.addWidget(reg_btn)
        cancel_btn = QPushButton("Отмена")
        cancel_btn.clicked.connect(self.reject)
        btns.addWidget(cancel_btn)
        layout.addLayout(btns)

    def _login(self):
        ok, result = self.account.login(self.login_user.text(), self.login_pass.text())
        if ok:
            self.user_data = result
            self.accept()
        else:
            QMessageBox.critical(self, "Ошибка", result)

    def _register(self):
        if self.reg_pass.text() != self.reg_pass2.text():
            QMessageBox.warning(self, "Ошибка", "Пароли не совпадают")
            return
        ok, result = self.account.register(self.reg_user.text(), self.reg_email.text(), self.reg_pass.text())
        if ok:
            self.user_data = result
            self.accept()
        else:
            QMessageBox.critical(self, "Ошибка", result)
